tests/unit/test_charm_2.py

```diff
@@ -630,3 +698,101 @@ def test_plugin_reconciliation(self):
+
+        self.harness.update_config(plugin_config)
+
+        self.assertEqual(
+            self.patch.get_active_plugins(db_host="config_db_host", db_name="config_db_name"),
+            {plugin},
+            f"{plugin} should be activated after {plugin_config_keys} being set"
+        )
+        self.assertEqual(
+            self.patch.get_options(db_host="config_db_host", db_name="config_db_name"),
+            excepted_options,
+            "akismet options should be set correctly"
+        )
+
+        if additional_check_after_install is not None:
+            additional_check_after_install()
+
+        self.harness.update_config({k: "" for k in plugin_config})
+        self.assertEqual(
+            self.patch.get_active_plugins(db_host="config_db_host", db_name="config_db_name"),
+            set(),
+            f"{plugin} should be deactivated after {plugin_config_keys} being reset"
+        )
+        self.assertEqual(
+            self.patch.get_options(db_host="config_db_host", db_name="config_db_name"),
+            {},
+            f"{plugin} options should be removed after {plugin_config_keys} being reset"
+        )
+
+    def test_akismet_plugin(self):
+        """
+        arrange: after peer relation established and database ready
+        act: update akismet plugin configuration
+        assert: plugin should be activated with WordPress options being set correctly, and plugin
+            should be deactivated with options removed after config being reset
+        """
+        self._standard_plugin_test(
+            plugin="akismet",
+            plugin_config={"wp_plugin_akismet_key": "test"},
+            excepted_options={
+                "akismet_strictness": "0",
+                "akismet_show_user_comments_approved": "0",
+                "wordpress_api_key": "test"
+            }
+        )
+
+    def test_openid_plugin(self):
+        """
+        arrange: after peer relation established and database ready
+        act: update openid plugin configuration
+        assert: plugin should be activated with WordPress options being set correctly, and plugin
+            should be deactivated with options removed after config being reset
+        """
+        self._standard_plugin_test(
+            plugin="openid",
+            plugin_config={
+                "wp_plugin_openid_team_map":
+                    "site-sysadmins=administrator,site-editors=editor,site-executives=editor"
+            },
+            excepted_options={
+                'openid_required_for_registration': '1',
+                'openid_teams_trust_list': 'a:3:{i:1;O:8:"stdClass":4:{s:2:"id";i:1;s:4:"team";s:14:"site-sysadmins";s:4:"role";s:13:"administrator";s:6:"server";s:1:"0";}i:2;O:8:"stdClass":4:{s:2:"id";i:2;s:4:"team";s:12:"site-editors";s:4:"role";s:6:"editor";s:6:"server";s:1:"0";}i:3;O:8:"stdClass":4:{s:2:"id";i:3;s:4:"team";s:15:"site-executives";s:4:"role";s:6:"editor";s:6:"server";s:1:"0";}}'
```

Comment:
Out of curiosity, how was this derived?

Reply:
This is PHP array serialization, currently I am copying all related methods from the original branch. But I have already developed a better way to do this `wp option update <json> --format=json`. This better method is only used in openstack swift plugin, but I will use this to replace all old methods like this line you mentioned.